tsconfig.json

```diff
@@ -0,0 +1,47 @@
+{
+  "compilerOptions": {
```

Comment:
BTW we could re-use `tsconfigs` from RNTL, as it recently went from JS/Flow to TS, and users didn't complain about these configs: 
- https://github.com/callstack/react-native-testing-library/blob/main/tsconfig.json
- https://github.com/callstack/react-native-testing-library/blob/main/tsconfig.release.json

@Ne3l wdyt?

Reply:
If you prefer it in order to simplify configs on your side we can do it but i wonder why we need the DOM on the react-native side https://github.com/callstack/react-native-testing-library/blob/main/tsconfig.json#L9

Reply:
It should simplify things since Jest Native is almost always used together with RNTL.

DOM lib there could be due to couple of reasons, most probably access to `window`. Let's use RNLT config but drop that one particular thing: `lib: ['ES2020', 'DOM']` => `lib: ['ES2020']`

Reply:
https://github.com/testing-library/jest-native/pull/101/commits/470b57dfd33821519249b2a84d0b0add51834c66 

Use RNLT config with some modifications:
- Include `"files": ["./extend-expect.d.ts"],` to support jest-native on our own test
- Add  `"exclude": ["node_modules", "android", "ios"],`
- Add `"setup-test.ts"` to support jest-native on our own test
- By default we have `noEmit` as false. In the typescript command we pass `--no-emit` flag and when we build it will emit.
- Removed DOM from lib